integration-libs/opf/components/opf-verify-payment/opf-payment-verification.component.ts

```diff
@@ -0,0 +1,90 @@
+/*
+ * SPDX-FileCopyrightText: 2023 SAP Spartacus team <spartacus-team@sap.com>
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+import { Component, OnDestroy, OnInit } from '@angular/core';
+import { ActivatedRoute } from '@angular/router';
+import { GlobalMessageService, GlobalMessageType } from '@spartacus/core';
+import {
+  KeyValuePair,
+  OpfCheckoutFacade,
+  OpfConfig,
+  OpfPaymentVerificationResponse,
+  OpfPaymentVerificationResult,
+  OpfPaymenVerificationUrlInput,
+} from '@spartacus/opf/root';
+import { OrderFacade } from '@spartacus/order/root';
+import { of, Subscription, throwError } from 'rxjs';
+import { switchMap } from 'rxjs/operators';
+import { OpfUrlHandlerService } from '../opf-url-handler.service';
+
+@Component({
+  selector: 'cx-opf-verify-payment',
+  templateUrl: './opf-payment-verification.component.html',
+})
+export class OpfPaymentVerificationComponent implements OnInit, OnDestroy {
+  subscription?: Subscription;
+
+  constructor(
+    protected route: ActivatedRoute,
+    protected orderFacade: OrderFacade,
+    protected opfCheckoutService: OpfCheckoutFacade,
+    protected config: OpfConfig,
+    protected globalMessageService: GlobalMessageService,
+    protected opfUrlHandlerService: OpfUrlHandlerService
+  ) {}
+
+  ngOnInit(): void {
+    this.subscription = of(this.route.routeConfig?.data?.cxRoute as string)
+      .pipe(
+        switchMap((cxRoute) => {
+          return cxRoute === 'paymentVerificationResult'
+            ? this.route.queryParams
+            : throwError('CANCEL URL RETURNED BY PSP');
+        }),
+        switchMap((params) => {
+          if (!params) return throwError('No params');
+
+          const paramsList: KeyValuePair[] =
+            this.opfUrlHandlerService.convertParamsToKeyValuePairs(params);
+
+          const paymentSessionId =
+            this.opfUrlHandlerService.findFromKeyValuePairs(
+              OpfPaymenVerificationUrlInput.PAYMENT_SESSION_ID,
+              paramsList
+            );
+          if (!paymentSessionId) return throwError('No paymentSessionId found');
+
+          return this.opfCheckoutService.verifyPayment(paymentSessionId, {
+            responseMap: [...paramsList],
+          });
+        }),
+        switchMap((response: OpfPaymentVerificationResponse) => {
+          return response?.result === OpfPaymentVerificationResult.AUTHORIZED
+            ? this.orderFacade.placeOrder(true)
+            : throwError('UNAUTHORIZED payment from OPF Adapter');
```

Comment:
One thing related to the throwing errors this way. You should be aware that is not optimal way for the case if we want to display specific error message to the customer.
We mosty use this approach in services to point out that some logic is an error.

However in terms of showing a message to the customer in UI the recommended way is using `GlobalMessageService`.

Example:
```
  protected displayError(): void {
    this.globalMessageService.add(
      {
        key: 'opf.yourErrorTranslationKey',
      },
      GlobalMessageType.MSG_TYPE_ERROR
    );
  }
```

The main advantage of this solution is the possibility of passing a translation key. By doing this, customers will always receive an error message translated to the language they are currently using.